pkg/imgpkg/bundle/bundle.go

```diff
@@ -19,48 +21,116 @@ const (
 )
 
 type Bundle struct {
-	plainImg     *plainimg.PlainImage
-	imgRetriever ctlimg.ImagesMetadata
+	plainImg      *plainimg.PlainImage
+	imgRetriever  ctlimg.ImagesMetadata
+	numSubBundles int
 }
 
 func NewBundle(ref string, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
-	return &Bundle{plainimg.NewPlainImage(ref, imagesMetadata), imagesMetadata}
+	return &Bundle{plainimg.NewPlainImage(ref, imagesMetadata), imagesMetadata, 0}
 }
 
 func NewBundleFromPlainImage(plainImg *plainimg.PlainImage, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
-	return &Bundle{plainImg, imagesMetadata}
+	return &Bundle{plainImg, imagesMetadata, 0}
 }
 
 func (o *Bundle) DigestRef() string { return o.plainImg.DigestRef() }
 func (o *Bundle) Repo() string      { return o.plainImg.Repo() }
 func (o *Bundle) Tag() string       { return o.plainImg.Tag() }
 
-func (o *Bundle) Pull(outputPath string, ui ui.UI) error {
+func (o *Bundle) Pull(outputPath string, ui goui.UI, recursive bool) error {
+	return o.pull(recursive, outputPath, "", map[string]bool{}, ui)
```

Comment:
order of vars for Pull and pull could be made consistent to make it less surprising when reading.

Reply:
lol I was actually going for a plot twist in this story. I'll update this.